pkg/network/protocols/postgres/telemetry.go

```diff
@@ -14,25 +14,39 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-const numberOfBuckets = 4
+const (
+	numberOfBuckets        = 10
+	bucketRange            = 15
+	belowBufferBucketCount = 3
+)
 
 // Telemetry is a struct to hold the telemetry for the postgres protocol
 type Telemetry struct {
 	metricGroup *libtelemetry.MetricGroup
 
-	// exceededQueryLengthBuckets holds the counters for the different buckets of exceeding query length quires
-	exceededQueryLengthBuckets [numberOfBuckets]*libtelemetry.Counter
+	// queryLengthBuckets holds the counters for the different buckets of by the query length quires
+	queryLengthBuckets [numberOfBuckets]*libtelemetry.Counter
 	// failedTableNameExtraction holds the counter for the failed table name extraction
 	failedTableNameExtraction *libtelemetry.Counter
 	// failedOperationExtraction holds the counter for the failed operation extraction
 	failedOperationExtraction *libtelemetry.Counter
 }
 
 // createQueryLengthBuckets initializes the query length buckets
+// Bucket 1: <= 33   query length
+// Bucket 2: 34 - 48 query length
+// Bucket 3: 49 - 63 query length
+// Bucket 4: 64 - 78 query length
```

Comment:
that's problematic
the buffer limit (64) should be the end of a bucket. Otherwise, we won't be able to distinguish between queries in range (64 long) and not truncated queries (65 and more)

Reply:
changed the startSize to include 64 in bucket 3